titus_isolate/event/kubernetes_opportunistic_window_publisher.py

```diff
@@ -51,41 +60,68 @@ def __get_node(self) -> V1Node:
-            now = datetime.utcnow()
-            if now < self.__get_timestamp(item['spec']['window']['end']):
-                return True
-        return False
+        with self.__lock:
+            log.debug('is active: oppo list: %s', json.dumps(self.__opportunistic_resources))
+            for item in self.__opportunistic_resources.values():
+                log.debug('checking for window: %s', json.dumps(item))
+                now = datetime.utcnow()
+                if now < self.__get_timestamp(item['object']['spec']['window']['end']):
+                    return True
+            return False
 
     def cleanup(self):
-        oppo_list = self.__get_scoped_opportunistic_resources()
-        log.debug('cleanup: oppo list: %s', json.dumps(oppo_list))
-        clean_count = 0
-        check_secs = self.__config_manager.get_float(OVERSUBSCRIBE_CLEANUP_AFTER_SECONDS_KEY,
-                                                     DEFAULT_OVERSUBSCRIBE_CLEANUP_AFTER_SECONDS)
-        if check_secs <= 0:
-            log.info('configured to skip cleanup. opportunistic resource windows will not be deleted.')
-            return 0
-        for item in oppo_list['items']:
-            check_time = datetime.utcnow() + timedelta(seconds=-1*check_secs)
-            if check_time < self.__get_timestamp(item['spec']['window']['end']):
-                continue
-            log.debug('deleting: %s', json.dumps(item))
-            delete_opts = V1DeleteOptions(grace_period_seconds=0, propagation_policy='Foreground')
-            resp = self.__custom_api.delete_namespaced_custom_object(version=OPPORTUNISTIC_RESOURCE_VERSION,
-                                                                     group=OPPORTUNISTIC_RESOURCE_GROUP,
-                                                                     plural=OPPORTUNISTIC_RESOURCE_PLURAL,
-                                                                     namespace=OPPORTUNISTIC_RESOURCE_NAMESPACE,
-                                                                     name=item['metadata']['name'],
-                                                                     body=delete_opts)
-            log.debug('deleted: %s', json.dumps(resp))
-            clean_count += 1
-
-        return clean_count
+        with self.__lock:
+            log.debug('cleanup: oppo list: %s', json.dumps(self.__opportunistic_resources))
+            clean_count = 0
+            check_secs = self.__config_manager.get_float(OVERSUBSCRIBE_CLEANUP_AFTER_SECONDS_KEY,
+                                                         DEFAULT_OVERSUBSCRIBE_CLEANUP_AFTER_SECONDS)
+            if check_secs <= 0:
+                log.info('configured to skip cleanup. opportunistic resource windows will not be deleted.')
+                return 0
+            for item in self.__opportunistic_resources.values():
+                check_time = datetime.utcnow() + timedelta(seconds=-1 * check_secs)
```

Comment:
how about just utcnow() - timedelta(check_sec) ?

Reply:
yup, that's better.  done.